src/daemon/daemon.cpp

```diff
@@ -2452,6 +2465,18 @@ try
             operation = [this, delay_minutes = std::chrono::minutes(request->time_minutes())](
                             VirtualMachine& vm) { return this->shutdown_vm(vm, delay_minutes); };
 
+        operation = [op = std::move(operation)](VirtualMachine& vm) {
+            if (vm.current_state() == VirtualMachine::State::unavailable)
+            {
+                mpl::log(mpl::Level::info,
+                         vm.vm_name,
+                         "Ignoring stop since instance is unavailable.");
+                return grpc::Status::OK;
```

Comment:
Shouldn't this be handled by the VM instead? If you try to stop a VM that is already stopped, it knows to ignore it. Shouldn't that be the case with this one too? Perhaps with the log, which could be present in both cases IMO.